sdk/core/core/internal/az_precondition_internal.h

```diff
@@ -43,23 +43,22 @@
 
 az_precondition_failed_fn az_precondition_failed_get_callback();
 
+AZ_INLINE void _az_precondition(bool condition)
```

Comment:
I haven't been following the other threads too closely. Can you summarize why we are turning macros into inline functions (maybe in the PR description)? How do we verify this is doing what it wants us to do.

Reply:
This is a draft and an experiment. I am linking to the PR where Jeff had this idea in comments.
The idea is that with inline functions, the compiler is smart enough to remove the code that has no effect, so we turn macros into inline functions, and they are more readable and better debuggable.